core/modules/cart/helpers/preparePaymentMethodsToSync.ts

```diff
@@ -10,15 +10,18 @@ const preparePaymentMethodsToSync = (
   const paymentMethods = [...currentPaymentMethods]
   const uniqueBackendMethods = []
 
-  for (const backendPaymentMethod of backendPaymentMethods) {
-    if (isPaymentMethodNotExist(backendPaymentMethod, currentPaymentMethods)) {
-      const backendMethod = {
-        ...backendPaymentMethod,
-        is_server_method: true
-      }
+  // Check if the variable is iterable before to enter in the for...of loop
+  if (typeof backendPaymentMethods[Symbol.iterator] === 'function') {
```

Comment:
Interface says it is `PaymentMethod[]` so please just check it with `Array.isArray`. This approach would be also proper and developers not familiar with Symbols, iterators, and typeof operator will understand the code.